include/swift/Basic/LLVM.h

```diff
@@ -89,4 +89,16 @@ namespace swift {
   using llvm::raw_ostream;
 } // end namespace swift
 
+/// Declares a const method with the given name and arguments intended for
+/// dumping. Methods declared with this macro should never be used except in the
+/// debugger; they are deprecated and marked used.
+#define DECLARE_DUMP_NAMED(name, ...) \
+LLVM_ATTRIBUTE_DEPRECATED(void name(__VA_ARGS__) const LLVM_ATTRIBUTE_USED, \
+                          "only for use in the debugger")
+
+/// Declares a const method named `dump` with the given arguments.
+/// Methods declared with this macro should never be used except in the
+/// debugger; they are deprecated and marked used.
+#define DECLARE_DUMP(...) DECLARE_DUMP_NAMED(dump, __VA_ARGS__)
```

Comment:
These don't belong in swift/Basic/LLVM.h; this header is only for bringing LLVM things into Swift's namespace. Maybe just make a new one?

Reply:
My thought was that this is wrapping `LLVM_ATTRIBUTE_DEPRECATED` and `LLVM_ATTRIBUTE_USED`, but that's kind of an implementation detail—I'll look for a better home for it.